### ISSUE TITLE
Axis orientation mismatch?

### ISSUE THREAD

**Issue by m053m716** [Spokhim/CMU_EMGSL#1]
Hey Pok,

Managed to get the `EMG_pipeline.ipynb` running (on Windows, even!) with code on the [`dev/Max`](https://github.com/Spokhim/CMU_EMGSL/tree/dev/Max) branch. Big props on your writing cross-platform-supported notebook!

One problem: I noticed the printout for which axes are expected to match to what directions from the `R_Forearm.dcm` did not seem to match the 3D plot that gets generated at the beginning of the script:
> Upside should be facing towards us.  i.e. Palmside in positive y-direction.  Electrodes go along x dimension.  Z dimension is proximal to distal.

![image](https://github.com/user-attachments/assets/5d422c9d-90d7-4a97-87f7-94b32e7cf70f)

As you can see, it looks like instead, palm-side in this plot would be one of the `z` axis directions, electrode rings would be oriented in the `xy` plane, and `y` axis would indicate proximal to distal.  

I updated Markdown in the notebook to reflect the orientations as it appeared, but then when I got to the superposition of the electrodes on the individual frame slices it looks like something is wrong with orientation of the arm relative to coordinate basis for the electrodes (see attached). 

![image](https://github.com/user-attachments/assets/f554ec54-72f9-451c-8688-50fe9c6a927f)

Any quick fix for this?

Cheers,
Max

**Comment by m053m716** [comment]
Update: looks like you were second-guessing yourself on orientation, I remember you mentioning something about that on one of the last days you were here. I think you actually did have it correct (or else, the correction is not working as intended and needs to be re-examined). I've applied a temporary fix by commenting the following from cell 20 (`dev/Max` version) in `EMG_Pipeline.ipynb`:
```
# Load electrode positions
electrode_pos = np.load('Data/256simparm_electrode_pos.npy')

# Load forward model
mne_fwd = mne.read_forward_solution('Data/simp_arm_2mm-fwd.fif')
fwd = mne_fwd['sol']['data']
pos = mne_fwd['source_rr'] 

xscaling, yscaling, zscaling = (2e-3, 2e-3, 2e-3) # Is the distance between dipoles in source space #np.repeat(np.abs(np.sum(np.diff(pos[0], axis = 0))),3)  

# Transformations specific to dataset
# Need to remove some electrodes - get rid of the middle ones
fwd = np.vstack((fwd[:64,:], fwd[192:,:]))
electrode_pos = np.vstack((electrode_pos[:64,:], electrode_pos[192:,:]))
# Adjust axes as it was saved in ["X", "Z", "Y"]
# electrode_pos = electrode_pos[:,[0,2,1]]
# pos = pos[:,[0,2,1]]
```
Now the slices look as expected  (example attached):  

![image](https://github.com/user-attachments/assets/b0ffaff0-e22a-42f8-b2b5-ca859b0d9835)

Leaving the Issue open so you see it in case you want to double-check the correction; otherwise feel free to close the issue as you see fit.
Best,
Max

**Comment by Spokhim** [comment]
Hi Max,

Yes this bug stems from an issue I had when saving the models in Blender, as by default Blender saved it in as Y in the upwards direction and -Z as the forwards direction.  Whereas my original intention (and our previous work) had Z in the upwards direction.   Hence, the need for the change of axes. 

I ended up exporting the models in the appropriate dimensions, regenerating the inputs which I provided to you.
This also meant fixing up the re-orientation code as part of commit: `ee2ec6ae523452d2f47ac345ace66ab05343f138` - "Adjusted mesh orientation to no longer require axes changes"
In which case, yep, the only difference in `Emg_pipeline.ipynb` is commenting out the change of axes which you have done so.   There were also adjustments in `fwd_BEM_MNE.ipynb` which is needed for the generation of the fwd model provided, but not a concern if a new forward model does not need to be generated.  

Tl;dr - Bug is a result of using newer model files in intended orientations, but with code one commit behind (oops I should've told you I adjusted the code as well as the input files!) which fixed previous orientation mistakes.  Closing.
